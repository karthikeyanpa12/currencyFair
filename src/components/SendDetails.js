import React, { Component } from 'react';
import { connect } from 'react-redux';

export class SendDetails extends Component {
    render() {
        const { rate, fee, deliveryDate } = this.props.transaction;
        const receiver = (2000.00 * rate) - fee;

        return (
            <div className="sent-container">
                <div className="sent-details">
                    <div>
                        <div className="sent-border">Sending Details</div>
                        <span className="align-left"> You send </span> <span className="float-right">£2000.00</span>
                    </div>
                    <div>
                        <div className="sent-border receiver"><span>Receving Details</span> <span className="float-right color-blue"> As of right now</span></div>
                        <div className="align-left"><span >Rate</span><span className="float-right">{rate}</span></div>
                        <div className="align-left"><span >Fee</span><span className="float-right">${fee}</span></div>
                        <div className="align-left"><span >Delivery date</span><span className="float-right">{deliveryDate}</span></div>
                        <div className="align-left"><span >Recipent gets</span><span className="float-right">£{receiver}</span></div>
                    </div>
                    <div className="sent-border saved">
                        You save £66.19 compared your bank!
                </div>
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        transaction: state.transaction
    };
};


export default connect(mapStateToProps)(SendDetails);

