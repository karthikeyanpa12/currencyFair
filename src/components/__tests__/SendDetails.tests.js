import React from 'react';
import "babel-polyfill";
import {SendDetails} from '../SendDetails';
import { shallow } from 'enzyme';

describe('Send Details', () => {
    const transaction = { rate: 0.5, fee: 2.5, deliveryDate: '29-sep-2019' }
    const tree = shallow(<SendDetails transaction={transaction} />);

    test('renders correctly', async () => {
        expect(tree).toMatchSnapshot();
    });

});
