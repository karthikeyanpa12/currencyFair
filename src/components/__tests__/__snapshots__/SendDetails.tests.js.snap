// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Send Details renders correctly 1`] = `ShallowWrapper {}`;
